src/bartholomew.rs

```diff
@@ -43,6 +44,29 @@ pub fn render(req: Request) -> Result<Response> {
     // Load the site config.
     let mut config: template::SiteInfo = toml::from_slice(&std::fs::read(CONFIG_FILE)?)?;
 
+    let mut dynamic_templates: HashMap<String, (String, String)> = HashMap::new();
+    if let Some(val) = &config.dynamic_templates {
```

Comment:
A `unwrap_or_default` would allow you to avoid the `if let` and do a map... once you had wrestled the borrow checker for the soul of humanity.  Along the lines of 
```
let dynamic_templates = &config
    .dynamic_templates
    .unwrap_or_default()
    .map(|t| (t.field1.to_owned, Something { t.field2, t.field3 })
    .collect::<HashMap<_, _>>();
```